pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -174,6 +192,19 @@ pub mod pallet {
 	pub(crate) type RelayerList<T: Config> =
 		StorageDoubleMap<_, Blake2_128Concat, Domain, Blake2_128Concat, DomainAddress, ()>;
 
+	/// Storage for outbound messages that will be processed `on_idle`.
+	#[pallet::storage]
+	pub(crate) type OutboundMessageQueue<T: Config> = StorageNMap<
+		_,
+		(
+			NMapKey<Blake2_128Concat, Domain>,
+			NMapKey<Blake2_128Concat, T::AccountId>,
+			NMapKey<Blake2_128Concat, T::Message>,
+		),
+		(),
+		ValueQuery,
+	>;
```

Comment:
Q: Why is the message part of the key instead of the value?

Reply:
If we were to use a double map with k1: domain, k2: sender, we would then need to keep a vector of messages. I wanted to avoid doing that and having to mingle with that vector every time we process something. This approach is also easier to iterate on IMO.